cms/djangoapps/contentstore/views/component.py

```diff
@@ -57,7 +62,19 @@ def _advanced_component_types():
     """
     Return advanced component types which can be created.
     """
-    return [c_type for c_type in ADVANCED_COMPONENT_TYPES if c_type not in settings.DEPRECATED_ADVANCED_COMPONENT_TYPES]
+    if XBlockDeprecatedAdvancedComponentConfig:
+        deprecated_xblock_types = XBlockDeprecatedAdvancedComponentConfig.disabled_block_types()
+    else:
+        deprecated_xblock_types = []
+
+    # Merge settings list with one in the admin config;
+    if hasattr(settings, 'DEPRECATED_ADVANCED_COMPONENT_TYPES'):
+        deprecated_xblock_types = list(deprecated_xblock_types)
+        deprecated_xblock_types.extend(
+            c_type for c_type in settings.DEPRECATED_ADVANCED_COMPONENT_TYPES if c_type not in deprecated_xblock_types
+        )
```

Comment:
All this logic should be inside `XBlockDeprecatedAdvancedComponentConfig.disabled_block_types()`.